hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -60,14 +61,39 @@ public HoodieCommitArchiveLog(HoodieWriteConfig config, FileSystem fs) {
      */
     public boolean archiveIfRequired() {
         List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
+        boolean success = true;
         if (commitsToArchive.iterator().hasNext()) {
             log.info("Archiving commits " + commitsToArchive);
             archive(commitsToArchive);
-            return deleteCommits(commitsToArchive);
+            success = deleteInstants(commitsToArchive);
         } else {
             log.info("No Commits to archive");
-            return true;
         }
+        return success & deleteOtherInstants();
+    }
+
+    private boolean deleteOtherInstants() {
```

Comment:
yes. We need some work before we can archive the instants (move archive log to be based on hoodielogformat). #165 captures that work. This is a PR to make COW not leave .clean files. More work needed which will be part of #165